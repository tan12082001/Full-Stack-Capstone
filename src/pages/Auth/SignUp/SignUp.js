import React from 'react';
import { NavLink } from 'react-router-dom';
import SignUpForm from './SignupForm';

const SignUp = () => (
  <div className='signup-page-outer'>
    <h1 id='signup-title'>Signup page</h1>
    <div className='signup-form-outer'>
      <SignUpForm />
    </div>
    <p>Have an existing account ?</p>
    <NavLink to="/account/signin">
      Log In
    </NavLink>
  </div>
);

export default SignUp;
