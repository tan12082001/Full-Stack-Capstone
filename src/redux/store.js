import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import authenticationReducer from './authentication/authenticationSlice';

const store = configureStore({
  reducer: {
    authencation: authenticationReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
