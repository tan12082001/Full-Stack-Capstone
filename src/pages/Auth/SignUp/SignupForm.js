import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import FormComponent from '../../../components/Form/FormComponent';
import { SignUpSchema, signUpInitialValues } from '../../../models/signup.model';
import FormSubmitButton from '../../../components/Button/FormSubmitButton';
import { TextInputField } from '../../../components/Form/FormField';
import HideableTextFormField from '../../../components/Form/HideableTextFormField';
import { registerUser } from '../../../redux/thunk';

const SignUpForm = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const status = useSelector((state) => state.authentication.status);
  const handleSubmit = (values) => {
    dispatch(registerUser(values));
  };

  if (status === 'succeeded') {
    navigate('/account/signin');
  }

  return (
    <FormComponent
      initialValues={signUpInitialValues}
      schema={SignUpSchema}
      onSubmit={handleSubmit}
      className="sign-up-form"
    >
      <TextInputField label="Name" name="username" placeholder="eg. Jane Doe" className="form-username" id="signup-username" />
      <TextInputField label="Email" name="email" placeholder="eg. example@example.com" className="form-email" id="signup-email" />
      <HideableTextFormField label="Password" name="password" className="password_block" id="signup-password" />
      <HideableTextFormField label="Confirm Password" name="password_confirmation" className="password_block" id="signup-confirm-password" />
      <FormSubmitButton type="submit" className="sign-up-submit">
        Sign Up
      </FormSubmitButton>
    </FormComponent>
  );
};

export default SignUpForm;
