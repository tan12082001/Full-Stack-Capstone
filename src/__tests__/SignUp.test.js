import '@testing-library/jest-dom/';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/store';
import SignUp from '../pages/Auth/SignUp/SignUp';

describe('SignUp component', () => {
  test('should render SignUp page', () => {
    const component = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <SignUp />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(component).toMatchSnapshot();
  });
});
