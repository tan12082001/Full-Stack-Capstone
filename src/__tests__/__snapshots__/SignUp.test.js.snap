// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component should render SignUp page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="signup-page-outer"
      >
        <h1
          id="signup-title"
        >
          Sign up
        </h1>
        <p
          class="user-headline"
        >
          Hello there! Register and start managing your application
        </p>
        <div
          class="signup-form-outer"
        >
          <form
            action="#"
            class="sign-up-form css-i51pdu"
          >
            <div
              class="css-1oq5xzy"
            >
              <label
                class="css-1mcvbvm"
              >
                Name
              </label>
              <div
                class="css-z2ctse"
              >
                <input
                  class=" css-4qk13k"
                  name="username"
                  placeholder="eg. Jane Doe"
                  value=""
                />
              </div>
            </div>
            <div
              class="css-1oq5xzy"
            >
              <label
                class="css-1mcvbvm"
              >
                Email
              </label>
              <div
                class="css-z2ctse"
              >
                <input
                  class=" css-4qk13k"
                  name="email"
                  placeholder="eg. example@example.com"
                  value=""
                />
              </div>
            </div>
            <div
              class="password_block css-1oq5xzy"
            >
              <label
                class="css-1mcvbvm"
              >
                Password
              </label>
              <div
                class="css-z2ctse"
              >
                <input
                  class="css-4qk13k"
                  name="password"
                  type="password"
                  value=""
                />
                <div
                  class="css-1xom5oo"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-uqi4wi-MuiSvgIcon-root"
                    data-testid="VisibilityOffIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="css-1oq5xzy"
            >
              <label
                class="css-1mcvbvm"
              >
                Confirm Password
              </label>
              <div
                class="css-z2ctse"
              >
                <input
                  class="css-4qk13k"
                  name="password_confirmation"
                  type="password"
                  value=""
                />
                <div
                  class="css-1xom5oo"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-uqi4wi-MuiSvgIcon-root"
                    data-testid="VisibilityOffIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <button
              class="sign-up-submit css-180h7py"
              type="submit"
            >
              Sign Up
            </button>
          </form>
        </div>
        <p>
          Have an existing account ?
        </p>
        <a
          class=""
          href="/account/signin"
        >
          Log In
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="signup-page-outer"
    >
      <h1
        id="signup-title"
      >
        Sign up
      </h1>
      <p
        class="user-headline"
      >
        Hello there! Register and start managing your application
      </p>
      <div
        class="signup-form-outer"
      >
        <form
          action="#"
          class="sign-up-form css-i51pdu"
        >
          <div
            class="css-1oq5xzy"
          >
            <label
              class="css-1mcvbvm"
            >
              Name
            </label>
            <div
              class="css-z2ctse"
            >
              <input
                class=" css-4qk13k"
                name="username"
                placeholder="eg. Jane Doe"
                value=""
              />
            </div>
          </div>
          <div
            class="css-1oq5xzy"
          >
            <label
              class="css-1mcvbvm"
            >
              Email
            </label>
            <div
              class="css-z2ctse"
            >
              <input
                class=" css-4qk13k"
                name="email"
                placeholder="eg. example@example.com"
                value=""
              />
            </div>
          </div>
          <div
            class="password_block css-1oq5xzy"
          >
            <label
              class="css-1mcvbvm"
            >
              Password
            </label>
            <div
              class="css-z2ctse"
            >
              <input
                class="css-4qk13k"
                name="password"
                type="password"
                value=""
              />
              <div
                class="css-1xom5oo"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-uqi4wi-MuiSvgIcon-root"
                  data-testid="VisibilityOffIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="css-1oq5xzy"
          >
            <label
              class="css-1mcvbvm"
            >
              Confirm Password
            </label>
            <div
              class="css-z2ctse"
            >
              <input
                class="css-4qk13k"
                name="password_confirmation"
                type="password"
                value=""
              />
              <div
                class="css-1xom5oo"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-uqi4wi-MuiSvgIcon-root"
                  data-testid="VisibilityOffIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <button
            class="sign-up-submit css-180h7py"
            type="submit"
          >
            Sign Up
          </button>
        </form>
      </div>
      <p>
        Have an existing account ?
      </p>
      <a
        class=""
        href="/account/signin"
      >
        Log In
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
